doc/internals/interoperability.rst

```diff
@@ -0,0 +1,45 @@
+.. _interoperability:
+
+Interoperability of Xarray
+==========================
+
+Xarray is designed to be extremely interoperable, in many orthogonal ways.
+Making xarray as flexible as possible is the common theme of most of the goals on our development :ref:`roadmap`.
+
+This interoperability comes via a set of flexible abstractions into which the user can plug in. The current full list is:
+
+- :ref:`Custom file backends <add_a_backend>` via the :py:class:`~xarray.backends.BackendEntrypoint` system,
+- Numpy-like :ref:`"duck" array wrapping <internals.duckarrays>`, which supports the `Python Array API Standard <https://data-apis.org/array-api/latest/>`_,
+- :ref:`Chunked distributed array computation <internals.chunkedarrays>` via the :py:class:`~xarray.core.parallelcompat.ChunkManagerEntrypoint` system,
+- Custom :py:class:`~xarray.Index` objects for :ref:`flexible label-based lookups <internals.custom indexes>`,
+- Extending xarray objects with domain-specific methods via :ref:`custom accessors <internals.accessors>`.
+
+.. warning::
+
+    One obvious way in which xarray could be more flexible is that whilst subclassing xarray objects is possible, we
+    generally advise against it, instead recommending composition over inheritance. See the
```

Comment:
```suggestion
    One obvious way in which xarray could be more flexible is that whilst subclassing xarray objects is possible, we
    generally advise against it, instead recommending composition over inheritance. See the
```
```suggestion
    One obvious way in which xarray could be more flexible is that whilst subclassing xarray objects is possible, we
    currently don't support it in most transformations, instead recommending composition over inheritance. See the
```

- I have had success using it as a kinda "central object", with methods that operate on the dataset — which works quite well as long as no transformations are done 
- I would change "generally" to "currently" to highlight that it could be possible in the future...

Reply:
I'll just add a note with a link to issue https://github.com/pydata/xarray/issues/3980

Reply:
Oh I already link to it! I'll change the wording to be more encouraging to developers interested in helping.